清理codebase，添加条目收藏表

Jarvie8176 opened the pull request:
Closes https://github.com/bangumi/dev-env/issues/6

# 添加用户条目收藏记录表以及stub data
```
./sql/schema/chii_subject_interests.sql
./sql/data/data_chii_person_collects.sql
./sql/data/data_chii_subject_interests_treehole.sql
```
* <s>./un-processed/chii_person_collects.sql</s>: 已经由[bangumi_crt.sql#L129](https://github.com/bangumi/dev-env/blob/main/sql/schema/bangumi_crt.sql#L129)添加 
# 添加sql script读取顺序设置
现在已经配置了`99.sql`，添加`100.sql`会导致冲突（执行顺序：`1.sql, 10,sql, 100,sql, 11.sql, ...`）

* 配置文件：`./sql_script_load_order.txt`
* 按行执行文件内的sql script
* 允许空行，会直接跳过

# 清理`docker-compose.yml`
把sql script相关的volume mount移到了单独的loader里。通过`dist.py`生成需要执行的`public/dist.sql`
* 见 https://github.com/Jarvie8176/dev-env/tree/collection-tables#%E6%B5%8B%E8%AF%95

# 清理测试

<s>https://github.com/bangumi/dev-env/blob/main/dist.py#L23</s>
<s>https://github.com/bangumi/dev-env/blob/main/test.py#L88</s>
* 从`./sql_script_load_order.txt`读取已配置的sql script，不包含mysql本身的volume

<s>https://github.com/bangumi/dev-env/blob/main/test.py#L101</s>
* 后面已经计算双向set difference

# 清理`dist.py`
* 从`./sql_script_load_order.txt`读取已配置的sql script
* `./dist.py`输出的`dist.sql`内容不变
  * 每条添加进`dist.sql`的sql script会按顺序打印到stderr
![image](https://user-images.githubusercontent.com/3547456/147453713-df6f2f07-8b8e-4b12-b545-b7b59a2d6777.png)

Jarvie8176 committed:
place sql scripts

Jarvie8176 committed:
impl

see PR for details

Jarvie8176 committed:
Update readme.md

Jarvie8176 committed:
Update docker-compose.yaml

Remove extra whitespace

trim21 commented:
别加poetry了，就俩依赖

trim21 commented:
现在这样起docker-compose还得装个python，有点过于复杂

trim21 commented:
可以弄个dev分支构建SQL和compose文件，推到主分支

Jarvie8176 commented:
> 现在这样起docker-compose还得装个python，有点过于复杂

不加python的话有两个写法，`wget https://bangumi.github.io/dev-env/dist.sql`然后丢到/public，或者shell script里重命名所有sql文件再丢到/public。两种我都写过了

trim21 commented:
> > 现在这样起docker-compose还得装个python，有点过于复杂
> 
> 不加python的话有两个写法，`wget https://bangumi.github.io/dev-env/dist.sql`然后丢到/public，或者shell script里重命名所有sql文件再丢到/public。两种我都写过了

我比较希望直接clone然后docker-compose up 就能顺利启动

trim21 commented:
在dev分支用ci构建dist.sql和docker=compose.yaml然后推到默认分支吧。

Jarvie8176 committed:
sql文件配置

在dockerfile里重命名sql文件(zero padded)，然后丢到`/docker-entrypoint-initdb.d/`下面

更新sql以后docker-compose需要加上`--build`

Jarvie8176 committed:
Merge branch 'collection-tables' of https://github.com/Jarvie8176/dev-env into collection-tables

Jarvie8176 committed:
Update db.Dockerfile

remove debug message

trim21 commented:
你的compose file写错了，up不起来…

Jarvie8176 committed:
Update docker-compose.yaml

add context

upsuper commented:
@Trim21 我不太理解……这里有什么需要我review的吗？

trim21 commented:
@upsuper 手机上操作，点错了…
